### PR TITLE
Added type definitions for "editly"

### PR DESCRIPTION
**Pull request by jonasgeiler**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/editly/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,1063 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { fabric } from 'fabric';
+
+/**
+ * Edit and render videos.
+ *
+ * @param config - Config.
+ */
+declare function Editly(config: Editly.Config): Promise<void>;
```
Comment (peterblazejewicz) [inline types/editly/index.d.ts:13]: isn't config optional with some pros set to default as per convention?
https://github.com/mifi/editly/blob/1bd0d30acc7924b03fc5ea0615642b38343dad95/index.js#L21-L47
what's common usage scenario? (options no options)

Comment (Skayo) [reply types/editly/index.d.ts:13]: Well, if you don't specify any config, the `outPath` will be `undefined` which causes an error anyways. So I just thought I'll make it required right away...

Comment (andrewbranch) [reply types/editly/index.d.ts:13]: From the implementation:

```js
const isGif = outPath.toLowerCase().endsWith('.gif');

// ...

assert(outPath, 'Please provide an output path');
```

Oops 😛

**Review by andrewbranch — APPROVED** [review]

**Comment by Skayo** [comment]
Ready to merge

**Pull request merged**